pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -29,31 +27,36 @@ type ObjectWatcher interface {
 	Create(clusterName string, desireObj *unstructured.Unstructured) error
 	Update(clusterName string, desireObj, clusterObj *unstructured.Unstructured) error
 	Delete(clusterName string, desireObj *unstructured.Unstructured) error
-	NeedsUpdate(clusterName string, desiredObj, clusterObj *unstructured.Unstructured) (bool, error)
+	NeedsUpdate(clusterName string, oldObj, currentObj *unstructured.Unstructured) bool
```

Comment:
If there is any specific reason for changing the signature here? The `desired` represents the expectation state, it's might not an `old` one.

Reply:
The modified logic compares the resources before and after the update on the member cluster, so I suggest changing the naming here. "NeedUpdate" is actually not suitable to be placed here anymore, it needs to be optimized in the future.